trie/patriciaMerkleTrie.go

```diff
@@ -247,10 +247,27 @@ func (tr *patriciaMerkleTrie) Recreate(root []byte) (common.Trie, error) {
 	tr.mutOperation.Lock()
 	defer tr.mutOperation.Unlock()
 
-	return tr.recreate(root)
+	return tr.recreate(root, tr.trieStorage)
 }
 
-func (tr *patriciaMerkleTrie) recreate(root []byte) (*patriciaMerkleTrie, error) {
+// RecreateFromEpoch returns a new trie, given the options
+func (tr *patriciaMerkleTrie) RecreateFromEpoch(options common.RootHashHolder) (common.Trie, error) {
+	tr.mutOperation.Lock()
+	defer tr.mutOperation.Unlock()
+
+	if !options.GetEpoch().HasValue {
+		return tr.recreate(options.GetRootHash(), tr.trieStorage)
+	}
+
+	tsmie, err := newTrieStorageManagerInEpoch(tr.trieStorage, options.GetEpoch().Value)
```

Comment:
so we basically create a new instance of type trie storage manager on each RecreateFromEpoch call. Do we need to call the Close method on it on a defer function as to not leak resources?

Reply:
It does not create a new instance of trieStorageManager, but it rather wraps the existing one. I do not think a defer is needed.

Reply:
ok